linera-chain/src/manager.rs

```diff
@@ -424,51 +424,60 @@ where
         key_pair: Option<&KeyPair>,
         local_time: Timestamp,
         blobs: BTreeMap<BlobId, Blob>,
-    ) -> Result<Option<ValidatedOrConfirmedVote>, ViewError> {
+    ) -> Result<Option<ValidatedOrConfirmedVote>, ChainError> {
         let round = proposal.content.round;
+        if key_pair.is_some() && round < self.current_round() {
+            return Ok(None);
+        }
 
         // If the validated block certificate is more recent, update our locked block.
         if let Some(lite_cert) = &proposal.validated_block_certificate {
             if self
                 .locked
                 .get()
                 .as_ref()
-                .map_or(true, |locked| locked.round < lite_cert.round)
+                .map_or(true, |locked| locked.round() < lite_cert.round)
             {
                 let value = Hashed::new(ValidatedBlock::new(executed_block.clone()));
                 if let Some(certificate) = lite_cert.clone().with_value(value) {
-                    self.set_locked(certificate, blobs)?;
+                    self.set_locked(LockedBlock::Regular(certificate), blobs)?;
                 }
             }
+        } else if proposal.content.round.is_fast() {
```

Comment:
nitpick: you already have the `let round = ...` above. Either reuse it or remove the variable and inline.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3140/commits/3876f9ad5dd59e4c6e4121e8b9078590fa627c81.